torch/testing/_internal/common_distributed.py

```diff
@@ -378,22 +379,28 @@ def _join_processes(self, fn):
                 # Sleep to avoid excessive busy polling.
                 time.sleep(0.1)
             elapsed_time = time.time() - start_time
-            if fn in self.skip_return_code_checks:
+            if fn in self.skip_process_error_code_checks:
+                # Don't check process exit codes, since these tests expect
+                # processes to crash/exit uncleanly. Also skips timeout
+                # checking since that is not needed due to the crash.
+                self._check_no_test_errors(elapsed_time, check_process_exit_codes=False)
+            elif fn in self.skip_return_code_checks:
                 self._check_no_test_errors(elapsed_time)
             else:
                 self._check_return_codes(elapsed_time)
         finally:
             global TEST_SKIPS
             TEST_SKIPS = self.old_test_skips
 
-    def _check_no_test_errors(self, elapsed_time):
+    def _check_no_test_errors(self, elapsed_time, check_process_exit_codes=True):
         """
         Checks that we didn't have any errors thrown in the child processes.
         """
         for i, p in enumerate(self.processes):
-            if p.exitcode is None:
-                raise RuntimeError('Process {} timed out after {} seconds'.format(i, elapsed_time))
-            self.assertNotEqual(self.TEST_ERROR_EXIT_CODE, p.exitcode)
+            if check_process_exit_codes:
```

Comment:
nit: can the if statement go before the `for` loop so that we skip it when we don't need to check exit  codes?